core/vtk/ttkMarchingTetrahedra/ttkMarchingTetrahedra.cpp

```diff
@@ -0,0 +1,170 @@
+#include <ttkMacros.h>
+#include <ttkMarchingTetrahedra.h>
+#include <ttkUtils.h>
+
+#include <vtkCellData.h>
+#include <vtkDataArray.h>
+#include <vtkDataObject.h>
+#include <vtkDataSet.h>
+#include <vtkDoubleArray.h>
+#include <vtkFloatArray.h>
+#include <vtkIdTypeArray.h>
+#include <vtkInformation.h>
+#include <vtkNew.h>
+#include <vtkPointData.h>
+#include <vtkPolyData.h>
+#include <vtkSignedCharArray.h>
+#include <vtkUnsignedCharArray.h>
+#include <vtkUnsignedLongLongArray.h>
+
+vtkStandardNewMacro(ttkMarchingTetrahedra);
+
+ttkMarchingTetrahedra::ttkMarchingTetrahedra() {
+  this->setDebugMsgPrefix("MarchingTetrahedra");
+  SetNumberOfInputPorts(1);
+  SetNumberOfOutputPorts(1);
+}
+
+int ttkMarchingTetrahedra::FillInputPortInformation(int port,
+                                                    vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkDataSet");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkMarchingTetrahedra::FillOutputPortInformation(int port,
+                                                     vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkDataObject::DATA_TYPE_NAME(), "vtkPolyData");
+    return 1;
+  }
+  return 0;
+}
+
+template <typename vtkArrayType, typename vectorType>
+void setArray(vtkArrayType &vtkArray, vectorType &vector) {
```

Comment:
if this is a helper, this should go into an anonymous namespace, at the top of the file not to be confused with method classes.